Documentation/devel/distribution-point.md

```diff
@@ -0,0 +1,174 @@
+
+#### OCI Image distribution(s)
+
+This is an Indirect distribution point.
+
+OCI images can be retrieved using a Docker registry but in future the OCI image spec will define one or more own kinds of distribution starting from an image name (with additional tags/labels).
+
+#### OCI Image layout
+
+This is a Direct distribution point.
+
+This can fetch an image starting from a [OCI image layout][oci layout] format. The 'location' can point to:
+
+* A single file archive
+* A local directory based layout
+* A remote directory based layout
+* Other types of locations
+
+This will probably end up being the final distribution used by the above OCI image distributions (like ACIArchive is the final distribution point for the Appc distribution point):
+
+* `cimd:oci-image-layout:v=0:file%3A%2F%2Fabsolute%2Fpath%2Fto%2Ffile?ref=refname`
+* `cimd:oci-image-layout:v=0:https%3A%2F%2Fdir%2F?ref=refname`
+
+Since the OCI image layout can provide multiple images selectable by a ref, one needs to specify which ref to use in the archive distribution URI (see the above ref query parameter). Since distribution only covers one image, it is not possible to import all refs with a single distribution URI.
+
+**TODO(sgotti)**: Define if oci-image-layout. It should internally handle both archive and directory based layouts or use two different distributions or a query parameter the explicitly define the layout (to avoid guessing if the URL points to a single file or to a directory).*
+
+**Note** Considering [this OCI image spec README section][oci image spec readme], the final distribution format will probably be similar to the Appc distribution. There is a need to distinguish their User Friendly string (prepending an appc: or oci: ?).
+
+To sum it up:
+
+| Distribution Point | Type     | URI Format                                                                | Final Distribution Point |
+|--------------------|----------|---------------------------------------------------------------------------|--------------------------|
+| Appc               | Direct   | `cimd:appc:v=0:name?label01=....&label02=...`                             | ACIArchive               |
+| Docker             | Direct   | `cimd:docker:v=0:[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG&#124;@DIGEST]` | <none>                   |
+| ACIArchive         | Indirect | `cimd:aci-archive:v=0:ArchiveURL?query...`                                |                          |
+| OCI                | Direct   | `cimd:oci:v=0:TODO`                                                       | OCIImageLayout           |
+| OCIImageLayout     | Indirect | `cimd:oci-image-layout:v=0:URL?ref=...`                                   |                          |
+
+## User friendly distribution strings
+
+Since the distribution URI can be complex one needs to help the user to request an image via some user friendly string. Rkt already has these kind of available input image strings (now mapped to an AppImageType):
```

Comment:
The distribution URI can be long and complex. It is helpful to have a friendly string for users to request an image with. Rkt supports a couple of image string input styles. These are mapped to an `AppImageType`: